### ISSUE TITLE
Bad markdown parsing of underscores in tables and lists

### ISSUE THREAD

**Issue by plaindocs** [wooorm/remark-lint#103]
Hi,

I'm using the [Coala](https://github.com/coala/bear-docs/blob/master/docs/MarkdownBear.rst) implementation of Remark, so not sure if issues should be raised here or there. If I'm in the wrong place please let me know.

I'm noticing a number of erroneous parsing / fixes, especially on subsequent runs of the linter. For example:

```diff
-| repos_count | number of repositories                              |
+| repos\_count | number of repositories                              |
```
in a table. 

And also 

```diff
-  2\.1.3 and 2.1.5.
+  2.1.3 and 2.1.5.
```

**Comment by wooorm** [comment]
So after running with remark, you get the added lines (green ones)?

**Comment by plaindocs** [comment]
Yes. Also does not seem to be 100% consistent, if you run a second time it escapes again, or removes the escape. Is this a difference between cli and parser?

**Comment by wooorm** [comment]
Weird. Well, removing the escapes is intended behaviour: they’re not needed.

Adding them back again is a bug. Could you show me those input / output examples?

**Comment by plaindocs** [comment]
Right. If I run with the default config on `https://raw.githubusercontent.com/travis-ci/docs-travis-ci-com/gh-pages/user/languages/javascript-with-nodejs.md` the first pass adds escapes, and the second removes them.

**Comment by giovanism** [comment]
@wooorm, is this fixed yet? Remark does still escape the dots in numbered list in [my case](https://gist.github.com/giovanism/1c5a5b63765874114da789298708eedf)

**Comment by wooorm** [comment]
It works the way you intend if you enable commonmark.
It works everywhere if there is a blank line between the paragraph and the list.
I think, now GitHub switched to support “commonmark-like” markdown, we should always enable this behaviour.
So for now, either enable commonmark mode, or, would you be interested in working on a PR to allow this behaviour everywhere?

**Comment by giovanism** [comment]
> It works the way you intend if you enable commonmark.
> It works everywhere if there is a blank line between the paragraph and the list.

Thanks. This wasn't really obvious for me. coala doesn't have way to pass json setting to remark. If commonmark is the way to go, I will gladly make the changes to my documents.
